js/src/components/contact-information/phone-number-card.js

```diff
@@ -105,67 +128,114 @@ function PhoneNumberContent( {
 	);
 }
 
+function EditPhoneNumberCard( { phoneNumber, onPhoneNumberChange } ) {
+	const { loaded, data } = phoneNumber;
+	const phoneNumberContent = loaded ? (
+		<>
+			<CardDivider />
+			<PhoneNumberContent
+				initCountry={ data.country }
+				initNationalNumber={ data.nationalNumber }
+				onPhoneNumberChange={ onPhoneNumberChange }
+			/>
+		</>
+	) : (
+		<AppSpinner />
+	);
+
+	return (
+		<AccountCard
+			{ ...basePhoneNumberCardProps }
+			description={ __(
+				'Please enter a phone number to be used for verification.',
+				'google-listings-and-ads'
+			) }
+		>
+			{ phoneNumberContent }
+		</AccountCard>
+	);
+}
+
+/**
+ * Renders phone number data in Card UI and is able to edit.
+ *
+ * @param {Object} props React props.
+ * @param {PhoneNumber} props.phoneNumber Phone number data.
+ * @param {boolean} [props.isPreview=false] Whether to display as preview UI.
+ * @param {boolean|null} [props.initEditing=null] Specify the inital UI state. This prop would be ignored if `isPreview` is true.
+ *     `true`: initialize with the editing UI.
+ *     `false`: initialize with the viewing UI.
+ *     `null`: determine the initial UI state according to the `data.isValid` after the `phoneNumber` loaded.
+ * @param {Function} [props.onEditClick] Called when clicking on "Edit" button.
+ *     If this callback is omitted, it will enter edit mode when clicking on "Edit" button.
+ * @param {onPhoneNumberChange} [props.onPhoneNumberChange] Called when inputs of phone number are changed in edit mode.
+ */
 export default function PhoneNumberCard( {
 	phoneNumber,
-	isPreview,
-	initEditing,
+	isPreview = false,
+	initEditing = null,
 	onEditClick,
 	onPhoneNumberChange = noop,
 } ) {
-	const [ isEditing, setEditing ] = useState( initEditing );
 	const { loaded, data } = phoneNumber;
-
-	const editButton = isEditing ? null : (
-		<AppButton
-			isSecondary
-			onClick={ () => {
-				if ( onEditClick ) {
-					onEditClick();
-				} else {
-					setEditing( true );
-				}
-			} }
-		>
-			{ __( 'Edit', 'google-listings-and-ads' ) }
-		</AppButton>
+	const [ isEditing, setEditing ] = useState(
+		isPreview ? false : initEditing
 	);
 
-	let description = null;
-	let phoneNumberContent = null;
+	useEffect( () => {
+		if ( loaded && isEditing === null ) {
+			setEditing( ! data.isValid );
+		}
+	}, [ loaded, data.isValid, isEditing ] );
```

Comment:
💅  It took me a while to figure out this relates to your JSDoc comment `` "determine the initial UI state according to the `data.isValid` after the `phoneNumber` loaded." ``. Would be good to have a comment to explain this, or even refactor it.

Reply:
Added a comment by https://github.com/woocommerce/google-listings-and-ads/pull/934/commits/63460f79b5c69866d7559ce8af57e36e26681fbd.